public/docs/ts/latest/cookbook/aot-compiler.jade

```diff
@@ -0,0 +1,177 @@
+include ../_util-fns
+
+:marked
+    Angular components consist of a mix of standard html and Angular syntax (e.g. `ngIf`, `ngFor`). 
+    
+    Expresions like `ngIf` and `ngFor` are specific to Angular, so there is no way for the browser to execute them directly.
+    
+    Before the browser can render the application, Angular specific code and templates have to be converted to regular executable JavaScript. We refer to this step as compilation.
+
+    By default compilation is executed by the browser at runtime, during what is called Just in Time compilation (JIT). It's called "Just in Time" since compilation happens on the fly as the application loads.
+
+    The downside to JIT compilation is performance. Views take longer to render because of the compilation step. It also forces us to download the Angular compiler along with our application code since we will need the compiler at runtime.  
+    
+    In this cookbook we discuss how we can improve performance by doing compilation at build time instead. We call this process Ahead of Time compilation (AOT). 
+<a id="toc"></a>
+:marked
+   ## Table of contents   
+
+   [Ahead of Time vs Just in Time](#aot-jit)
+
+   [Compilation](#compilation)
+
+   [Tree Shaking](#tree-shaking)
+
+   [Bootstrapping](#bootstrapping)
+
+.l-main-section
+<a id="aot-jit"></a>
+:marked
+    ## Ahead of Time (AOT) vs Just in Time (JIT)
+
+    AOT and JIT compilation use the same exact compiler. The difference is, AOT happens at build time, JIT happens at runtime.  
+
+    ### Why do AOT compilation?
+
+    The performance improvement from doing AOT compilation can be significant for two reasons:
+
+    *Faster rendering*
+
+    When doing AOT compilation the browser downloads a pre-compiled version of the application. This means the browser can start rendering the application immediately, without delays from doing JIT compilation first.
+
+    The compiler will also inline external files like html templates and css files, which means we no longer need separate ajax requests to load external files at run time.
```

Comment:
sounds like 3 reasons.  downloads a pre-compiled version, inlines html thus reducing requests for HTML, and small angular framework download